src/plots/plots.js

```diff
@@ -366,6 +366,10 @@ plots.supplyDefaults = function(gd) {
     newFullLayout._basePlotModules = [];
     newFullLayout._subplots = emptySubplotLists();
 
+    // for traces to request a default rangeslider on their x axes
+    // eg set `_requestRangeslider.x2 = true` for xaxis2
+    newFullLayout._requestRangeslider = {};
```

Comment:
It would be nice to add a mock with a ohlc or candlestick on a x-axis other than `layout.xaxis` to :lock: this down.

Reply:
the mock in https://github.com/plotly/plotly.js/pull/2561/commits/0b7541e34d63ce74cddb5d243a652b2df0430b70 🔒 this.